catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/HiveTableOperations.java

```diff
@@ -215,8 +219,84 @@ private org.apache.hadoop.hive.metastore.api.Partition toHivePartition(
   }
 
   @Override
-  public boolean dropPartition(String partitionName) {
-    throw new UnsupportedOperationException();
+  public boolean dropPartition(String partitionName, boolean ifExists)
+      throws NoSuchPartitionException {
+    try {
+      // check the partition exists
+      if (!ifExists) {
+        table
+            .clientPool()
+            .run(c -> c.getPartition(table.schemaName(), table.name(), partitionName));
+      }
+
+      // get all partitions that will be deleted
+      Table hiveTable = table.clientPool().run(c -> c.getTable(table.schemaName(), table.name()));
+      List<org.apache.hadoop.hive.metastore.api.Partition> partitions =
+          table
+              .clientPool()
+              .run(
+                  c ->
+                      c.listPartitions(
+                          table.schemaName(),
+                          table.name(),
+                          getFilterPartitionList(hiveTable, partitionName),
+                          (short) -1));
+
+      // delete partitions iteratively
+      for (org.apache.hadoop.hive.metastore.api.Partition partition : partitions) {
+        table
+            .clientPool()
+            .run(
+                c ->
+                    c.dropPartition(
+                        partition.getDbName(),
+                        partition.getTableName(),
+                        partition.getValues(),
+                        false));
+      }
+    } catch (UnknownTableException e) {
+      if (!ifExists) {
+        throw new NoSuchTableException(
+            e, "Hive table %s does not exist in Hive Metastore", table.name());
+      }
+
+    } catch (NoSuchObjectException e) {
+      if (!ifExists) {
+        throw new NoSuchPartitionException(
+            e, "Hive partition %s does not exist in Hive Metastore", partitionName);
+      }
+
+    } catch (TException | InterruptedException e) {
+      throw new RuntimeException(
+          "Failed to get partition "
+              + partitionName
+              + " of table "
+              + table.name()
+              + "from Hive Metastore",
+          e);
+    }
+    return true;
+  }
+
+  @Override
+  public boolean dropPartitions(List<String> partitionNames, boolean ifExists)
+      throws NoSuchPartitionException, UnsupportedOperationException {
+    if (partitionNames.size() > 1) {
+      throw new UnsupportedOperationException("Only one partition is supported");
```

Comment:
If not support drop multiple partitions, why implement this interface? I mean, this is mis-leading. How about do that with a loop?

Reply:
It's a complex problem to considerate what should do if some partitions of the partition list drop failed.
So according to the result of discussion with @mchades and @jerryshao, this version we both provide rest api about dropPartitions and dropPartition, but dropPartitions only support drop one partition, like addPartitions only support add one partition in this version 
<img width="866" alt="image" src="https://github.com/datastrato/gravitino/assets/58762426/45944e33-deac-4b99-9ded-0a2d44bc94da">

Reply:
OK，I'll remove dropPartitions in this PR